cmd/bffutil/main.go

```diff
@@ -0,0 +1,377 @@
+	for i := 0; i < c.NArg(); i++ {
+		var org *models.Organization
+		if org, err = GetOrg(c.Args().Get(i)); err != nil {
+			return cli.Exit(err, 1)
+		}
+		orgs = append(orgs, org)
+	}
+
+	if len(orgs) == 1 {
+		return printJSON(orgs[0])
+	}
+	return printJSON(orgs)
+}
+
+func addCollab(c *cli.Context) (err error) {
+	// Collect the organization from the database
+	var org *models.Organization
+	if org, err = GetOrg(c.String("org")); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Fetch the user from auth0.
+	var user *management.User
+	if user, err = auth0.User.Read(c.String("user")); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Get the user appdata, roles, and permissions
+	appdata := &auth.AppMetadata{}
+	if err = appdata.Load(user.AppMetadata); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	var roles *management.RoleList
+	if roles, err = auth0.User.Roles(*user.ID); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	var permissions *management.PermissionList
+	if permissions, err = auth0.User.Permissions(*user.ID); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Ask if we should proceed
+	username, _ := auth.UserDisplayName(user)
+	fmt.Printf("User %s (%s, switch_organizations=%t) has appdata.OrgID %q\n", username, StringifyRoles(roles), HasPermission(auth.SwitchOrganizations, permissions), appdata.OrgID)
+	if !askForConfirmation(fmt.Sprintf("add user %q to organization %q?", username, org.ResolveName())) {
+		return cli.Exit("canceled at request of user", 0)
+	}
+
+	// Add the user to the new organization
+	collaborator := &models.Collaborator{
+		Email:    *user.Email,
+		UserId:   *user.ID,
+		Verified: *user.EmailVerified,
+	}
+
+	if err = org.AddCollaborator(collaborator); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	if err = db.UpdateOrganization(org); err != nil {
+		return cli.Exit(fmt.Errorf("could not update organization: %w", err), 1)
+	}
+
+	if HasPermission(auth.SwitchOrganizations, permissions) {
+		// If the user has the switch organizations permission then we just add them to
+		// the new organization but do not remove them from the old organization
+		appdata.AddOrganization(org.Id)
+	} else {
+		// If the user does not have the switch organizations permission, remove them
+		// from their current organization and add them to the new organization.
+		if appdata.OrgID != "" {
+			var curOrg *models.Organization
+			if curOrg, err = GetOrg(appdata.OrgID); err != nil {
+				return cli.Exit(err, 1)
+			}
+
+			curOrg.DeleteCollaborator(*user.Email)
+
+			if len(curOrg.Collaborators) == 0 {
+				// If there are no more collaborators in the current org, delete it.
+				if err = db.DeleteOrganization(curOrg.UUID()); err != nil {
+					return cli.Exit(fmt.Errorf("could not delete user's current organization: %w", err), 1)
+				}
+			} else {
+				if err = db.UpdateOrganization(curOrg); err != nil {
+					return cli.Exit(fmt.Errorf("could not update user's current organization: %w", err), 1)
+				}
+			}
```

Comment:
Sounds good to me, just wanted to make sure we thought of that; since we are primarily using this for TSPs I'm 100% okay with not attempting to change roles.